### PR TITLE
override public rate limiting through context

### PR DESCRIPTION
**Pull request by LukasJenicek**

### CODE REVIEW

**Review by klaidliadon — APPROVED** [review]
LGTM 👍

File: middleware/context.go [hunk 1]
```diff
@@ -18,12 +18,13 @@ func (k *contextKey) String() string {
...
-	ctxKeyAccessQuota   = &contextKey{"AccessQuota"}
-	ctxKeyComputeUnits  = &contextKey{"ComputeUnits"}
-	ctxKeyRateLimitSkip = &contextKey{"RateLimitSkip"}
-	ctxKeyTime          = &contextKey{"Time"}
-	ctxKeyResult        = &contextKey{"Result"}
+	ctxKeyAccessKey        = &contextKey{"AccessKey"}
+	ctxKeyAccessQuota      = &contextKey{"AccessQuota"}
+	ctxKeyComputeUnits     = &contextKey{"ComputeUnits"}
+	ctxKeyRateLimitSkip    = &contextKey{"RateLimitSkip"}
+	ctxOverridePublicLimit = &contextKey{"OverridePublicLimit"}
```
Comment (pkieltyka) [inline middleware/context.go:25]: for consistency of naming... we have `ctxKeyRateLimitSkip` and you introduced `ctxOverridePublicLimit` ... please follow consistent conventions..

therefore, your name should be .. `ctxKeyRateLimitPublicOverride`

File: middleware/context.go [hunk 2]
```diff
@@ -86,6 +87,16 @@ func IsSkipRateLimit(ctx context.Context) bool {
 	return ok
 }
 
+// WithOverridePublicLimit override rate limiting configuration.
+func WithOverridePublicLimit(ctx context.Context, limit int) context.Context {
```
Comment (pkieltyka) [inline middleware/context.go:91]: can you rename to `OverridePublicRateLimit` and the Get method too..

cuz its not clear what "limit" this is..

**Review by pkieltyka — COMMENTED** [review]
LGMT :) I didn't read the code super carefully in terms of logic, but if it offers overrides for some endpoints, then very good. I left some basic naming comments.

**Pull request closed without merging**